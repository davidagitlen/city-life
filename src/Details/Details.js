import React, { Component } from 'react';
import './Details.scss';

class Details extends Component{

  render() {
    console.log('in Details', this.props.props);
    const {country, population, latitude, longitude, fullName, timeZone} = this.props.props.details;
    const {attribution, images} = this.props.props.images;
    return(
      <>
        <div>
          <img src={images.web} alt={`${country}`} />
        </div>
        <section>
          <div>
            <p>{fullName}</p>
            <p><span>Population :</span> {population}</p>
            <p><span>Time Zone :</span> {timeZone}</p>
          </div>
          <div>
            <p>Map?</p>
          </div>
        </section>
      </>
    )
  }
}

export default Details;