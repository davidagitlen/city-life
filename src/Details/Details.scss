.Details{
  display: flex;
  flex-direction: column;
  width: 78%;
  margin: 10px auto;
  padding: 10px;

  & img {
    height: 100%;
    width: 100%;
  }
}


.Details-description {
  display: flex;
  flex-direction: row;
  justify-content: space-around;

  & div {
    margin: 25px 0px;
  }
  & p {
    font-size: 28px;
    margin: 15px 0px 15px 0px;
  }
}