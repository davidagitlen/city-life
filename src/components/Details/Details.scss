.Details{
  display: flex;
  flex-direction: column;
  width: 78%;
  margin: 10px auto;
  padding: 10px;
}

.Details-description {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  margin: 100px 0px 0px 0px;

  & div {
    margin: 25px 0px;
  }
  & p {
    font-size: 28px;
    margin: 15px 0px 15px 0px;
  }
}

.Details-image {
  -webkit-clip-path: inset(0px 10px);
  clip-path: inset(0px 10px);
}

.description-left {
  width: 100%;
}

.description-right {
  width: 50%;
}