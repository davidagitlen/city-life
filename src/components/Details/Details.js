import React from 'react';
import ReactHtmlParser from 'react-html-parser';
import './Details.scss';
import { Link } from 'react-router-dom';
import { convertFromString } from '../../util/dataCleaner';
import PropTypes from 'prop-types';

const Details = (props) => {
  const { index, cityData } = props;
  const {
    latitude, longitude, population, fullName, timeZone
  } = cityData.details[index];
  const { attribution, images } = cityData.images[index];
  const { summary } = cityData.description[index];
  const cleanTimeZone = timeZone.replace(/_/g, ' ');

  const mapURL = `https://api.mapbox.com/styles/v1/mapbox/light-v9/static/${longitude},${latitude},6,0,0/350x300?access_token=pk.eyJ1IjoiZGF2aWRhZ2l0bGVuIiwiYSI6ImNrMGs5NTNlcTA0dGkzY3MzdHZ3MTRiZGoifQ.ZtBWtklc66DFIgPC2CI_qg`;
  const titleAttribution = `Photo by ${attribution.photographer}, from ${attribution.site}. Original can be found at ${attribution.source}.`
  console.log('in Details, props: ', props)

  return(
    <div className='Details'>
      <Link to={`/`} className='back-btn'>â—€ back</Link>
      <div className='Details-image'>
        <img src={images.web} alt={`${fullName}`} title={titleAttribution} />
      </div>
      <div className='Details-description'>
        <div className='description-left'>
          {ReactHtmlParser(summary)}
          <p>{fullName}</p>
          <p><span>Population :</span> {population}</p>
          <p><span>Time Zone :</span> {cleanTimeZone}</p>
        </div>
        <div className='description-right'>
          <img src={mapURL} alt='' />
        </div>
      </div>
    </div>
  )
}

export default Details;

Details.propTypes = {
  cityData: PropTypes.object,
  index: PropTypes.number
}