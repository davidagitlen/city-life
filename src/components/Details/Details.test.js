import React from 'react';
import Details from './Details';
import { shallow } from 'enzyme';

describe('Details', () => {

  let wrapper, mockProps;
  
  beforeEach(() => {

    mockProps = {
        details: [{
          population: 1,
          latitude: 10,
          longitude: 10,
          fullName: 'Denver, Colorado!',
          timeZone: 'Denver!'
        }, {
          population: 2,
          latitude: 10,
          longitude: 10,
          fullName: 'Boulder, Colorado!',
          timeZone: 'Denver!'
        }],
        description: [
          { summary: 'A city!'},
          { summary: 'Another city!'}
        ],
        images: [{
          attribution: {},
          images: {
            mobile: 'http://blurg.com',
            web: 'http://blurg.com/web'
            }
          },
          {
          attribution: {},
          images: {
            mobile: 'http://blurg.com',
            web: 'http://blurg.com/web'
            }
          }],
        scores: [{ color: 'blue', name: 'Score', score_out_of_10: 10 }]
      };

    wrapper = shallow(
      <Details 
        cityData={mockProps}
        index={0}
      />
    );
  });

  it('should match the snapshot with data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

});