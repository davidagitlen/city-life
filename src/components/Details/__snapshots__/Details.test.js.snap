// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details should match the snapshot with data passed in correctly 1`] = `
<div
  className="Details"
>
  <Link
    className="back-btn"
    to="/"
  >
    â—€ back
  </Link>
  <div
    className="Details-image"
  >
    <img
      alt="Denver, Colorado!"
      src="http://blurg.com/web"
      title="Photo by undefined, from undefined. Original can be found at undefined."
    />
  </div>
  <div
    className="Details-description"
  >
    <div
      className="description-left"
    >
      A city!
      <p>
        Denver, Colorado!
      </p>
      <p>
        <span>
          Population :
        </span>
         
        1
      </p>
      <p>
        <span>
          Time Zone :
        </span>
         
        Denver!
      </p>
    </div>
    <div
      className="description-right"
    >
      <img
        alt=""
        src="https://api.mapbox.com/styles/v1/mapbox/light-v9/static/10,10,6,0,0/350x300?access_token=pk.eyJ1IjoiZGF2aWRhZ2l0bGVuIiwiYSI6ImNrMGs5NTNlcTA0dGkzY3MzdHZ3MTRiZGoifQ.ZtBWtklc66DFIgPC2CI_qg"
      />
    </div>
  </div>
</div>
`;
